datumaro/plugins/searcher.py

```diff
@@ -67,73 +32,56 @@ def img_normalize(image):
     return image
 
 
-def tokenize(texts: str, context_length: int = 77, truncate: bool = True):
-    checkpoint = "openai/clip-vit-base-patch32"
-    tokenizer = Tokenizer.from_pretrained(checkpoint)
-    tokens = tokenizer.encode(texts).ids
-    result = np.zeros((1, context_length))
-
-    if len(tokens) > context_length:
-        if truncate:
-            eot_token = tokens.ids[-1]
-            tokens = tokens[:context_length]
-            tokens[-1] = eot_token
-
-    for i, token in enumerate(tokens):
-        result[:, i] = token
-    return result
-
-
 class SearcherLauncher(OpenvinoLauncher):
     def __init__(
         self,
         description=None,
         weights=None,
         interpreter=None,
-        device=None,
         model_dir=None,
+        model_name=None,
         output_layers=None,
+        device=None,
     ):
-        model_name = "clip_visual_ViT-B_32"
-        if description or weights:
-            model_name = osp.splitext(description)[0]
-
-        url_folder = "https://storage.openvinotoolkit.org/repositories/datumaro/models/"
-        if not model_dir:
-            model_dir = os.path.join(os.path.expanduser("~"), ".cache", "datumaro")
-            if not osp.exists(model_dir):
-                os.makedirs(model_dir)
-
-        description = osp.join(model_dir, model_name + ".xml")
-        if not osp.exists(description):
-            cached_description_url = osp.join(url_folder, model_name + ".xml")
-            log.info('Downloading: "{}" to {}\n'.format(cached_description_url, description))
-            download_file(cached_description_url, description)
-
-        weights = osp.join(model_dir, model_name + ".bin")
-        if not osp.exists(weights):
-            cached_weights_url = osp.join(url_folder, model_name + ".bin")
-            log.info('Downloading: "{}" to {}\n'.format(cached_weights_url, weights))
-            download_file(cached_weights_url, weights)
-
-        if not interpreter:
-            openvino_plugin_samples_dir = get_samples_path()
-            interpreter = osp.join(openvino_plugin_samples_dir, "clip_ViT-B_32_interp.py")
-
-        super().__init__(description, weights, interpreter, device, model_dir, output_layers)
+        super().__init__(
+            description, weights, interpreter, model_dir, model_name, output_layers, device
+        )
 
         self._device = device or "cpu"
         self._output_blobs = next(iter(self._net.outputs))
         self._input_blobs = next(iter(self._net.input_info))
 
+    def _tokenize(self, texts: str, context_length: int = 77, truncate: bool = True):
+        checkpoint = "openai/clip-vit-base-patch32"
+        tokenizer = Tokenizer.from_pretrained(checkpoint)
```

Comment:
I think that it would be good load `Tokenizer` checkpoint from the given model name. In addition, we need to store `tokenizer -> self.tokenizer` so that it is not loaded every time you tokenize.

Reply:
I corrected this. Thank you for pointing out this.